src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -806,3 +811,208 @@ def can_loop_be_parallelised(self, loop,
                 result = False
 
         return result
+
+# -------------------------------------------------------------------------
+    def can_loops_be_fused(self, loop1, loop2):
+        '''Function that verifies if two loops can be fused.
+
+        :param loop1: the first loop.
+        :type loop1: :py:class:`psyclone.psyir.nodes.Loop`
+        :param loop1: the first loop.
+        :type loop1: :py:class:`psyclone.psyir.nodes.Loop`
+
+        :return: whether the loops can be fused or not.
+        :rtype: bool
+
+        '''
+        # pylint: disable=too-many-locals
+        # This should only be called from loop_fuse_trans, which
+        # has done tests for loop boundaries (depending on domain)
+
+        self._clear_messages()
+        assert isinstance(loop1, Loop)
+        assert isinstance(loop2, Loop)
+        vars1 = VariablesAccessInfo(loop1)
```

Comment:
I've forgotten how this works for LFRic. Does a kernel with a stencil  access (specified in its metadata) end up with a 'fake' array access created for the appropriate field argument?

Reply:
I'm sure I saw some emails with some responses to this review but GH isn't showing me anything now. Please could you comment on this comment?

Reply:
@arporter Dang, I was about to say I don't know what you mean, when I realised I have a 'review' pending (answering as 'review' means less emails). So, I'll click on finish review now, and hope that wonderful things will happen.